Brokerages/GDAX/OrderBook.cs

```diff
@@ -0,0 +1,173 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Brokerages.GDAX
+{
+    /// <summary>
+    /// Represents a full order book for a security.
+    /// It contains prices and order sizes for each bid and ask level.
+    /// The best bid and ask prices are also kept up to date.
+    /// </summary>
+    public class OrderBook
+    {
+        private readonly object _locker = new object();
+        private readonly Symbol _symbol;
+        private readonly SortedDictionary<decimal, decimal> _bids = new SortedDictionary<decimal, decimal>();
+        private readonly SortedDictionary<decimal, decimal> _asks = new SortedDictionary<decimal, decimal>();
+
+        /// <summary>
+        /// Event fired each time <see cref="BestBidPrice"/> or <see cref="BestAskPrice"/> are changed
+        /// </summary>
+        public event EventHandler<BestBidAskUpdatedEventArgs> BestBidAskUpdated;
```

Comment:
This is cool. Could see this type of event being super useful in algorithms if/when we support L2 data